### PR TITLE
Update and refine the enRoute website for the new R7 examples

### PR DESCRIPTION
**Pull request by timothyjward**
Squashed commits from Tim Ward and Richard Nicholson to ensure a clean merge and that GitHub pages are used throughout the history of master

Signed-off-by: Tim Ward <tim.ward@paremus.com>
Signed-off-by: Richard Nicholson <richard.nicholson@paremus.com>

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
+source "https://rubygems.org"
+
+# Hello! This is where you manage which Jekyll version is used to run.
+# When you want to use a different version, change it below, save the
+# file and run `bundle install`. Run Jekyll with `bundle exec`, like so:
+#
+#     bundle exec jekyll serve
+#
+# This will help ensure the proper Jekyll version is running.
+# Happy Jekylling!
```
Comment (bjhargrave) [inline Gemfile]: Not sure why these changes were made.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "osgi.enroute"]
```
[the remark this answers is not in the document]
Comment (timothyjward) [reply .gitmodules]: > Is this needed to build the site?

Yes - this is how we pull in code snippets without duplicating the source in the markdown files.

> Does github pages know to get git submodules?

Yes - it's described at https://help.github.com/articles/using-submodules-with-pages/

> Curious.

I couldn't find another way to reference them statically (the \<embed\> tag can't be used with GitHub URLs to do it live)

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,4 @@
 /_site/
 /.sass-cache/
 /.project
-/.bundle/
```
[the remark this answers is not in the document]
Comment (timothyjward) [reply .gitignore]: You can see the full history of the feature branch at timothyjward/osgi.enroute.site@677041944b8d50a696cff2f4de88c6cd13c0451c - it ended up in the weeds at a couple of points diverging significantly and using raw Jekyll as an attempt to pull in the source code snippets. 

Rather than pollute the history with lots of commits that wouldn't build in GitHub pages I got things building again with GitHub pages, squashed the commits, and then merged back correcting conflicts. I tried my best to avoid unnecessary deltas, but may not have caught everything.
